### PR TITLE
Added wrapper functions for REPLICATE() Function to fix issue with input containing multi-byte characters.

### PR DESCRIPTION
**Pull request by rohit01010**
### Description
Currently for string function `REPLICATE()` the existing definition takes input as type `TEXT` and return type is `VARCHAR`. This leads to incorrect output for input containing multi-byte characters with `NCHAR` and `NVARCHAR` datatype for input, as the return datatype should have been `NVARCHAR` for `NCHAR` and `NVARCHAR` datatype. This PR will fix this issue by adding wrapper functions with correct input and return datatype for `REPLICATE()` function.

Authored-by: Rohit Bhagat [rohitbgt@amazon.com](mailto:rohitbgt@amazon.com)
Signed-off-by: Rohit Bhagat [rohitbgt@amazon.com](mailto:rohitbgt@amazon.com)

### Issues Resolved

BABEL-862

### Test Scenarios Covered ###
* **Use case based -** YES

* **Boundary conditions -** YES

* **Arbitrary inputs -** YES

* **Negative test cases -** YES

* **Minor version upgrade tests -** YES

* **Major version upgrade tests -** YES

* **Performance tests -** NO

* **Tooling impact -** NO

* **Client tests -** NO

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/expected/replicate-vu-verify.out [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+abc?defghi??             abc?defghi??             abc?defghi??             abc?defghi??             
+~~END~~
+
+
+DECLARE @inputString NCHAR(25) = N'比尔·拉莫斯'
+SELECT replicate(@inputString, 4)
+GO
+~~START~~
+nvarchar
+??·???                   ??·???                   ??·???                   ??·???
```
Comment (shah-nirmit) [inline test/JDBC/expected/replicate-vu-verify.out:248]: Is this a known issue?

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.6.0--3.7.0.sql [hunk 2]
```diff
@@ -4890,6 +4890,126 @@ RETURNS OID
...
+$BODY$
+BEGIN
+    IF i < 0 THEN
+        RETURN NULL;
+    END IF;
+
+    RETURN PG_CATALOG.repeat(string, i);
+END;
+$BODY$
+LANGUAGE plpgsql IMMUTABLE PARALLEL SAFE;
```
Comment (shah-nirmit) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.6.0--3.7.0.sql]: STRICT? Also check fresh install files

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.6.0--3.7.0.sql [hunk 3]
```diff
@@ -4890,6 +4890,126 @@ RETURNS OID
...
```
Comment (shah-nirmit) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.6.0--3.7.0.sql]: STRICT?

**Review by shah-nirmit — CHANGES_REQUESTED** [review]

File: test/JDBC/expected/replicate-vu-verify.out (hunk 1 above)
Comment (rohit01010) [reply test/JDBC/expected/replicate-vu-verify.out:248]: Yes

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.6.0--3.7.0.sql (hunk 2 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.6.0--3.7.0.sql]: Done

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.6.0--3.7.0.sql (hunk 3 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.6.0--3.7.0.sql]: Done

**Review by shah-nirmit — APPROVED** [review]
LGTM

**Review by shah-nirmit — APPROVED** [review]

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**